unity-renderer/Assets/DCLPlugins/ECS7/CanvasPainter/Tests/CanvasPainterUITransformIntegrationTest.cs

```diff
@@ -0,0 +1,333 @@
+﻿using System.Collections;
+using System.Linq;
+using DCL.CRDT;
+using DCL.ECS7.UI;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+using DCL.Helpers;
+using DCL.Models;
+using DCLPlugins.ECS7.TestsUtils;
+using NSubstitute;
+using NSubstitute.Extensions;
+using NUnit.Framework;
+using Tests;
+using UnityEngine;
+using UnityEngine.TestTools;
+using UnityEngine.UI;
+using UnityEngine.UIElements;
+
+namespace DCL.ECS7.Tests
+{
+    public class CanvasPainterUITransformIntegrationTest 
+    {
+        private RendererState rendererState;
+        private CanvasPainter canvasPainter;
+        private ECSComponentsManager componentsManager;
+        private ECS7ComponentsComposer componentsComposer;
+        private ECS7TestUtilsScenesAndEntities testUtils;
+        private IWorldState worldState;
+        
+        [SetUp]
+        protected void SetUp()
+        {
+            worldState = Substitute.For<IWorldState>();
+            
+            rendererState = ScriptableObject.CreateInstance<RendererState>();
+            var componentFactory = new ECSComponentsFactory();
+            componentsComposer = new ECS7ComponentsComposer(componentFactory, Substitute.For<IECSComponentWriter>(), Substitute.For<IInternalECSComponents>());
+            componentsManager = new ECSComponentsManager(componentFactory.componentBuilders);
+            
+            testUtils = new ECS7TestUtilsScenesAndEntities(componentsManager);
+            canvasPainter = new CanvasPainter(DataStore.i.ecs7, rendererState,Substitute.For<IUpdateEventHandler>(), componentsManager, worldState);
+            rendererState.Set(true);
+            canvasPainter.rootNode.rootVisualElement.style.justifyContent = new StyleEnum<Justify>(Justify.Center);
+            canvasPainter.rootNode.rootVisualElement.style.alignContent = new StyleEnum<Align>(Align.Center);
+        }
+
+        [TearDown]
+        protected void TearDown()
+        {
+            componentsComposer.Dispose();
+            canvasPainter.Dispose();
+            DataStore.i.ecs7.scenes.Clear();
+            DataStore.i.ecs7.uiDataContainer = new UIDataContainer();
+        }
+
+        [Test]
+        public void DrawListOfItemCorrectly()
+        {
+            int parentId = 512;
+            int childIdStartIndex = 513;
```

Comment:
Curiosity: Have these numbers been choosen randomly? or have they these specific values for any reason?